kobo/rcS

```diff
@@ -62,8 +73,22 @@ fi
 /bin/mknod /dev/null c 1 3
 /bin/chmod 666 /dev/null
 
-/sbin/udevd --daemon
-/sbin/udevadm trigger
+# Make sure our /dev/ backup is valid (has a non-zero size).
+[ "$(ls -s /etc/udev.tgz | awk '{print $1}')" == 0 ] && rm -rf /etc/udev.tgz
+# If we don't know our specific platform, we rely on udev to create
+# /dev/ entries. Otherwise we use the /dev tree backup if it
+# exist. This is required for platforms that don't appear to have
+# complete udev support and ship a device tree instead (such as Nia).
+if [ $PLATFORM == freescale ] || [ ! -e /etc/udev.tgz ]; then
+	/sbin/udevadm control --env=STARTUP=1
+	/sbin/udevadm trigger
+	/sbin/udevadm settle --timeout=2
+	/sbin/udevadm control --env=STARTUP=
+	# Create a /dev/ tree backup to restore next time around.
+	[ $PLATFORM != freescale ] && tar cpzf /etc/udev.tgz /dev &
```

Comment:
This will leave a Zombie process around.

Reply:
This won't affect any supported model. The freescale platform only covers the original Kobo and the Kobo  Wifi, both models without touchscreen.

Anyways backgrounded jobs should be calling `waitpid()` on completion. No problem if you want to run this in the foreground either.

Reply:
Now that I read this again it doesn't make much sense. The `tar cpzf ...` will never execute as the condition isFreescale and isNotFreescale cannot be true at the same time.

Reply:
> Anyways backgrounded jobs should be calling `waitpid()` on completion.

Not quite - the parent process should be calling `waitpid()`, not the backgrounded job.
So, who's going to do that?

Reply:
Not familiarized with this repo to tell you. If rcS runs the entire time because XCSoar doesn't daemonize then either insert a `wait` or just don't background. Otherwise the child will be reaped when the script finish its execution.

**But** as it's currently wroten that chunk of code won't execute, as the precondition doesn't match the specific condition to untar device nodes.

Reply:
> If rcS runs the entire time

It does not. It `exec`s XCSoar's KoboMenu, replacing itself.

> because XCSoar doesn't daemonize

It does not, but it does not need to. The KoboMenu process just keeps running until you power the Kobo off. There's no point in daemonizing it.